### PR TITLE
To Do List Project: Add and remove functionality

### PR DESCRIPTION
**Pull request by JoyTaribagshaw**
> To Do list: Project: CRUD functionality Introduced Project Requirements:
> 
> * Removed all hardcoded items from the tasks array.
> * Created a new JavaScript file for the new functionality.
> * Implemented a function for adding a new task (add a new element to the array).
> * Implemented a function for deleting a task (remove an element from the array).
> * Implemented a function for editing task descriptions.
> * All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -15,18 +15,28 @@
...
       </form>
       <ul class="list-container">
+        <!-- <li class="list-item">
+          <div class="list-div">
+              <input type="checkbox" class="check" name="" id="">
+              <p>${myToDoLists.description}</p>
+                <input type="text" name="" id="" class="edit--input">
+                <i class="fa-solid fa-ellipsis-vertical" data--option=${myToDoLists.index}></i> -->
+          <!-- </div> -->
+      <!-- </li> -->
```
Comment (zieeco) [inline src/index.html]: - let's remove this commented code to make our codebase clean and maintainable.

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,107 @@
...
+  };
+
+  addTask = (description, completed = false) => {
+    const index = this.myToDoList.length + 1;
+    const newTask = { description, completed, index };
+    this.myToDoList = [...this.myToDoList, newTask];
+  };
+
+  removeTask = (index) => {
+    this.myToDoList = this.myToDoList.filter((list) => list.index !== +index);
```
Comment (zieeco) [inline src/index.js:76]: - Hey, let's try to reset the indexes after making any deletions. Currently, your indexes are not getting the correct value. see the image below to fix this.

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @JoyTaribagshaw,




Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice crud implementation. ✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions


- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html (hunk 1 above)
Comment (JoyTaribagshaw) [reply src/index.html]: Thank you so much for your review @zieeco , I have removed the comments.

File: src/index.js (hunk 2 above)
Comment (JoyTaribagshaw) [reply src/index.js:76]: Thank you so much for your review @zieeco , I have fixed the correct indexes.

**Review by rotimiazeez — APPROVED** [review]
Hi @JoyTaribagshaw 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the suggested changes 👍🏽
- Good App UI 👏🏽
- All required functionality implemented 👍🏽 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**